incrementalbuild/src/main/java/io/takari/incrementalbuild/spi/AbstractBuildContext.java

```diff
@@ -79,6 +85,11 @@ protected AbstractBuildContext(Workspace workspace, File stateFile,
       throw new NullPointerException();
     }
 
+    if (configuration.containsKey("mojo.parameter.failOnError")) {
```

Comment:
Clever but I don't like it:

* failOnError implementation is awkwardly split between explicit mojo parameter declaration and implicit behaviour implemented here.
* It retroactively applies to all mojos that happen to have `failOnError` parameter.
* Configuration is an opaque bucket of key/value pairs; it can be compared to other configuration instances, but individual parameters don't have any meaning (many are just SHA1 hashes of source configuration values).
* Proposed change assumes specific MojoConfigurationDigester implementation and particular parameter naming convention it happens to use

All in all, I think more direct `context.setFailOnError(boolean)` will be easier to understand and use.

Reply:
@ifedorenko I'll change to use  the method context.setFailOnError(failOnError) in the execute method in the AbstractCompileMojo instead read the configuration in the AbstractBuildContext